script.go

```diff
@@ -0,0 +1,581 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package statedb
+
+import (
+	"bufio"
+	"bytes"
+	"cmp"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io"
+	"iter"
+	"maps"
+	"os"
+	"regexp"
+	"slices"
+	"strings"
+	"text/tabwriter"
+	"time"
+
+	"github.com/cilium/hive"
+	"github.com/cilium/hive/script"
+	"gopkg.in/yaml.v3"
+)
+
+func ScriptCommands(db *DB) hive.ScriptCmdOut {
+	subCmds := map[string]script.Cmd{
+		"tables":     TablesCmd(db),
+		"show":       ShowCmd(db),
+		"export":     ExportCmd(db),
+		"cmp":        CompareCmd(db),
+		"insert":     InsertCmd(db),
+		"delete":     DeleteCmd(db),
+		"prefix":     PrefixCmd(db),
+		"lowerbound": LowerBoundCmd(db),
+	}
+	subCmdsList := strings.Join(slices.Collect(maps.Keys(subCmds)), ", ")
+	return hive.NewScriptCmd(
+		"db",
+		script.Command(
+			script.CmdUsage{
+				Summary: "Inspect and manipulate StateDB",
+				Args:    "cmd args...",
+				Detail: []string{
+					"Supported commands: " + subCmdsList,
+				},
+				// TODO detail to list the sub commands
+			},
+			func(s *script.State, args ...string) (script.WaitFunc, error) {
+				if len(args) < 1 {
+					return nil, fmt.Errorf("expected command (%s)", subCmdsList)
+				}
+				cmd, ok := subCmds[args[0]]
+				if !ok {
+					return nil, fmt.Errorf("command not found, expected one of %s", subCmdsList)
+				}
+				wf, err := cmd.Run(s, args[1:]...)
+				if err != nil {
+					s.Logf("usage: db %s %s\n", args[0], cmd.Usage().Args)
+				}
+				return wf, err
+			},
+		),
+	)
+}
+
+func TablesCmd(db *DB) script.Cmd {
+	return script.Command(
+		script.CmdUsage{
+			Summary: "Show StateDB tables",
+			Args:    "table",
+		},
+		func(s *script.State, args ...string) (script.WaitFunc, error) {
+			txn := db.ReadTxn()
+			tbls := db.GetTables(txn)
+			w := tabwriter.NewWriter(s.LogWriter(), 5, 4, 3, ' ', 0)
+			fmt.Fprintf(w, "Name\tObject count\tGraveyard count\tIndexes\tInitializers\tGo type\n")
```

Comment:
We could fairly easily keep some overall stats for each table and show them here (read and write rates etc.).
Also recording who has the current WriteTxn (lock) on the table and since when is easy to add.